Sign up module

Sartc00l opened the pull request:
Made user sign up module with associated user verification. 
Refactored the code 
(не понимаю что сюда писать.....)
User password reset in progress

Sartc00l committed:
User verification has been implemented

Sartc00l committed:
Refactor auth app

Created constants.py for error handling

Try to  make DRF best practices

views.py added mapping for errors, http responses only in views now.

Russian ->
Врефакторил код, старался соблюдать паттерны
Вывел константы в отдельный файл, чтоб не было циклических импортов

Убрал орфографические ошибки в названиях методов

Сброс пароля в процессе

bandirom commented on web/api/v1/auth_app/constants.py:
```diff
@@ -0,0 +1,36 @@
+
+цикличный импорт приветствуется 
+"""
+class AuthErrorMessage:
```
можешь здесь использовать Enum

Sartc00l replied:
Не понимаю как реализовать с eunm и gettext в связке

bandirom commented on web/api/v1/auth_app/serializers.py:
```diff
@@ -54,20 +45,23 @@ def validate(self, data: dict):
 class PasswordResetSerializer(serializers.Serializer):
     email = serializers.EmailField()
-
+    def validate_email(self,value):
+        if not User.objects.filter(email=value).exists():
+            msg = {'email':AuthErrorMessage.EMAIL_NOT_EXIST}#email not exi
+            raise serializers.ValidationError(msg)
```
обязательно нужно вернуть value, если валидация прошла успешно

bandirom commented on web/api/v1/auth_app/services.py:
```diff
@@ -89,23 +91,36 @@ def create_user(self, validated_data: dict):
-        user.save(update_fields=['is_active'])
-        return user
-
+    def verify_email_confirmation(key:str) :
```
если функция внутри класса, то должна иметь self или `@staticmethod/@classmethod`

Sartc00l replied:
Вывел логику подтверждения почты в другой класс VerifyEmailManager

bandirom commented on web/api/v1/auth_app/services.py:
```diff
@@ -89,23 +91,36 @@ def create_user(self, validated_data: dict):
-
+    def verify_email_confirmation(key:str) :
+        try:
+            user_id = signing.loads(key,max_age=ConfirmationEmailHandler.EXPIRATION_SECONDS)
```
Предлагаю создать классы `ConfirmationKeyManager` и `PasswordResetManager`, где ты опишешь функциональность шифрования и дешифрования токенов, чтобы не размазывать логику работы с этими процесами по разных классах

Sartc00l replied:
PasswordResetManager in progress. ConfrimationKeyManager -> VerifyEmailManager(вся логика подтверждения почты)

bandirom commented on web/api/v1/auth_app/views.py:
```diff
@@ -18,6 +21,21 @@
+    },
+    'link_expired':{
+        'detail': AuthErrorMessage.LINK_EXPIRED,
+        'status':status.HTTP_410_GONE
```
HTTP_400_BAD_REQUEST тоже будет норм

bandirom commented on web/api/v1/auth_app/views.py:
```diff
@@ -79,12 +96,27 @@ def post(self, request):
+        except ValidationError as e:
+            error_codes = e.get_codes()
+            #Для неизвестных ошибок это комментарии не от гпт....(честно)(а то запутаюсь)
+            error_cfg = ERRORS_MAPPING.get(
```
все ошибки должны быть известны

Sartc00l replied:
Видоизменил способ обработки ошибок

Sartc00l committed:
IN PROGRESS вывел логику подтверждения в отдельный класс

Sartc00l committed:
IN PROGRESS[VerifyEmailView] .gitigonre updt

добавил в gitignore dockerfile тк linux и macos + win имеют разные порты

Удалил набросок проверки от гпт тк дает OBJECT на фронт

Разобраться с работой enum

Sartc00l committed:
IN PROGRESS

Убрал Enum (сделал как было) .
Вернул передачу ошибок на фронт через маппер

Sartc00l commented on .gitignore:
Добавил в gitignore , тк на mac 127.0.0.1:8000 на линуксе нулевые

Sartc00l commented on web/auth_app/static/auth_app/js/reset-password.js:
```diff
@@ -0,0 +1,18 @@
+            console.log(errors)
+        }
+    })
+})
```
**In progress**

Sartc00l commented on web/auth_app/static/auth_app/js/verifie-email.js:
```diff
@@ -16,7 +16,7 @@ function verifyEmail() {
             var errors = xhr.responseJSON;
             if (errors) {
-                showMessage(errors)
+                showMessage(errors.detail)
```
Решение ошибки вывода на frontend [object Object]

Sartc00l committed:
in progress

Sartc00l committed:
in progress

bandirom commented on docker/dev/env/.db.env:
```diff
@@ -3,3 +3,6 @@ POSTGRES_PORT=5432
 POSTGRES_DB=develop
 POSTGRES_USER=develop
 POSTGRES_PASSWORD=develop
+SUPERUSER_EMAIL=example@email.com
+SUPERUSER_PASSWORD=secretp@ssword
+PROJECT_TITLE=MyProject
```
это ж переменные для базы даных. Они шарятся и в джангу и в постгрес

Sartc00l replied:
Это добро подтянулось с мака, проглядел....

bandirom commented on web/api/v1/auth_app/services.py:
```diff
@@ -89,27 +169,15 @@ def create_user(self, validated_data: dict):
-        logger.info("Сделал юзера!!!!!!!")
+        self._send_confirmation_email(user)
         return user
+    logger.info("User created") 
```
это куда ты логгер сдвинул?

Sartc00l replied:
Не заметил. Исправил

bandirom commented on web/api/v1/auth_app/services.py:
```diff
@@ -37,35 +43,109 @@ class CreateUserData(NamedTuple):
+            user = User.objects.get(id=user_id)
+            if user.is_active:
+                raise ValidationError(
+                    AuthErrorMessage.LINK_ALREADY_ACTIVATED,
```
логичнее не "линк" а "юзер" уже активирован

Sartc00l replied:
Сделано

bandirom commented on web/django_template.egg-info/SOURCES.txt:
```diff
@@ -6,12 +6,12 @@ api/urls.py
 api/v1/__init__.py
 api/v1/urls.py
 api/v1/auth_app/__init__.py
+api/v1/auth_app/constants.py
```
эти файлы не нужны в гите

Sartc00l replied:
С мака подтянулось......Проглядел

bandirom commented on web/django_template.egg-info/PKG-INFO:
```diff
@@ -3,7 +3,7 @@ Name: django-template
 Summary: Django template in docker with docker-compose
 Home-page: https://github.com/bandirom/django-template
-Author: Nazarii Romanchenko
+Author: Nazarii Romanchenko and dimasik
```
эти файлы не нужны в гите

Sartc00l replied:
подтянулось.........

Sartc00l committed:
IN PROGRESS

Сделал отправку письма , сделал frontend часть при переходе на ссылку

Sartc00l committed:
Again and again IN PROGRESS

Sartc00l committed:
feat(reset-password): implement password reset

Sartc00l committed:
style(login): Removed error in authentication

Sartc00l committed:
refactor(ui_resetpassword):extract showMessage

bandirom commented on web/auth_app/static/auth_app/js/login.js:
```diff
@@ -1,7 +1,12 @@
   $('#loginForm').submit(login);
-});
-
+});ß
```
ß

Sartc00l replied:
Убрано

bandirom commented on web/auth_app/static/auth_app/js/login.js:
```diff
@@ -1,7 +1,12 @@
+$(function() {
+  $('#registerBtn').on('click',function(e){
+    e.preventDefault()
+    window.location.href = '../register'
```
все редиректы делаются от корня, тоесть 
`/blog`, `/login`, `/profile`

Sartc00l replied:
http://127.0.0.1:8000/register/ <- путь регистрации же

bandirom commented on web/auth_app/static/auth_app/js/reset-password.js:
```diff
@@ -10,14 +10,8 @@ $('#forgotPasswordForm').submit(function (e) {
     })
     $('#pwdModal').modal('hide');
-    showMessage("Message has been delivered")
+    showMessage("Message has been delivered",'messageDiv')
+    clearMessage('messageDiv',10000)
```
сделай в  `showMessage` 3й аргумент, который будет устанавливать таймаут, если задан

bandirom commented on web/auth_app/static/auth_app/js/extends_js/ui-handler.js:
```diff
@@ -0,0 +1,23 @@
+
+function showMessage(message,messageBlock) {
+    const messageDiv = document.getElementById(messageBlock)
+    const tag = document.createElement("H1")
+    const informationContent = document.createTextNode(message)
+
+    tag.appendChild(informationContent)
+    messageDiv.appendChild(tag)
+
+    clearMessage(messageBlock,)
+}
+
+function clearMessage(messageSpace,seconds){
+    const timer = setInterval(()=>{
+        document.getElementById(messageSpace).textContent = ''
+    },seconds)
+}
+
+function timer(seconds) {
+    const timer = setInterval(() => { document.getElementById('timer').textContent = seconds < 0 
+        ? window.location.href = "../login" : seconds--; },seconds)
+}
```
Идеально для класса

Sartc00l replied:
Пока что шибко не хочется использовать ООП в js, если ссылаться буду чаще и больше, то докину ооп , пока что самый простой вариант -> вынести просто в отдельный файл и подключать его
